gdal/configure.ac

```diff
@@ -1492,23 +1490,14 @@ if test "x$PG_CONFIG" = "xno" ; then
   AC_MSG_RESULT([no])
 
 else
-  if test -d ${PG_CONFIG} ; then
-      AC_MSG_RESULT([no])
-      AC_MSG_ERROR([--with-pg argument is a directory.  It should be the path to the pg_config script, often somewhere like /usr/local/pgsql/bin/pg_config.])
-  fi
 
-  if test \! -x ${PG_CONFIG} ; then
-      AC_MSG_RESULT([no])
-      AC_MSG_ERROR([--with-pg argument is a not an executable file.  It should be the path to the pg_config script, often somewhere like /usr/local/pgsql/bin/pg_config.])
-  fi
-
-  AC_MSG_RESULT([yes])
-
-  AC_CHECK_LIB(pq,PQconnectdb,HAVE_PG=yes,HAVE_PG=no,-L`$PG_CONFIG --libdir`)
+  PKG_PROG_PKG_CONFIG([0.21])
+  PKG_CHECK_MODULES([PQ],[libpq > 9.1], [HAVE_PG=yes], [HAVE_PG=no])
 
   if test "${HAVE_PG}" = "yes" ; then
-    LIBS=-L`$PG_CONFIG --libdir`" -lpq $LIBS"
-    PG_INC=-I`$PG_CONFIG --includedir`" -I"`$PG_CONFIG --includedir-server`
+    PG_LIB="${PQ_LIBS}"
+    PG_INC="${PQ_CFLAGS}"
+    LIBS="${PG_LIB} ${LIBS}"
```

Comment:
could you add a check in a else clause so that if  "x$with_pg" = "xyes" (explicit ask from the user, to be opposed to implict try), AC_MSG_ERROR() is used to bail out.

Reply:
Sorry didn't catch where to put it and then what AC_MSG_ERROR() would do.

Reply:
Something along (untested !)
```
if test "${HAVE_PG}" = "yes" ; then
    LIBS=-L`$PG_CONFIG --libdir`" -lpq $LIBS"
    PG_LIB="${PQ_LIBS}"
    PG_INC=-I`$PG_CONFIG --includedir`" -I"`$PG_CONFIG --includedir-server`
    PG_INC="${PQ_CFLAGS}"
    LIBS="${PG_LIB} ${LIBS}"
else
     if "x$with_pg" = "xyes"; then
           AC_MSG_ERROR([--with-pg was requested, but libpq is not available])
     fi
fi
```

Reply:
Thanks make sense, I test that and adjust the PR once done